### PR TITLE
fix: remove the hazardous default.toml

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Remove the hazardous `default.toml` with `enable_recovery = false`

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

None

## Refer to a related PR or issue link (optional)
None

### CODE REVIEW

File: src/config/default.toml [hunk 1]
```diff
@@ -1,3 +0,0 @@
-[meta]
-disable_recovery = true
```
Comment (BugenZhao) [inline src/config/default.toml:2]: I guess we really want to disable recovery with RiseDev. 🤔

Comment (fuyufjh) [reply src/config/default.toml:2]: I don't think so 🤔 Actually for my cases I am used to `risedev d full` and it keeps my tables when rebooting.

**Review by BugenZhao — APPROVED** [review]

File: risedev.yml [hunk 2]
```diff
@@ -194,7 +193,7 @@ profile:
       - use: compactor
 
   docker-playground:
-    config-path: src/config/default.toml
+    config-path: src/config/playground.toml
```
Comment (xxchan) [inline risedev.yml:196]: btw, this gives docker playground idle-exit 🤔

**Comment by BugenZhao** [comment]
There are two senses of "recovery":

1. Reset the compute node and rebuild the streaming graph after checkpoint failure.
2. Meta service recovers the cluster from the meta store on re-launch with `./risedev full`.

I think for development usage, only `2` is necessary 🤔, while `1` may sometimes lead to unexpected/confusing behavior like #7247. cc @fuyufjh @yezizp2012 @xxchan What do you think?

**Comment by yezizp2012** [comment]
> There are two senses of "recovery":
> 
> 1. Reset the compute node and rebuild the streaming graph after checkpoint failure.
> 2. Meta service recovers the cluster from the meta store on re-launch with `./risedev full`.
> 
> I think for development usage, only `2` is necessary 🤔, while `1` may sometimes lead to unexpected/confusing behavior like #7247. cc @fuyufjh @yezizp2012 @xxchan What do you think?

+1, totally agree. `1` will definitely makes bugs during development more confusing.

**Comment by hzxa21** [comment]
> There are two senses of "recovery":
> 
> 1. Reset the compute node and rebuild the streaming graph after checkpoint failure.
> 2. Meta service recovers the cluster from the meta store on re-launch with `./risedev full`.
> 
> I think for development usage, only `2` is necessary 🤔, while `1` may sometimes lead to unexpected/confusing behavior like #7247. cc @fuyufjh @yezizp2012 @xxchan What do you think?

IMO, even for production usage, it is better to make unexpected/unsupported behaviors more obvious to ease debugging and guide user for correct usage. 

I am thinking whether we should have a "state" for a MV and mark it as "failed"/"paused" when non-recoverable things happen so that 1) this bad mv won't affect other non-dependent MVs; 2) user can  get meaningful error message when using the bad MV.

**Comment by fuyufjh** [comment]
> I think for development usage, only 2 is necessary 🤔, while 1 may sometimes lead to unexpected/confusing behavior like https://github.com/risingwavelabs/risingwave/issues/7247.

Totally agree. 

> I am thinking whether we should have a "state" for a MV and mark it as "failed"/"paused" when non-recoverable things happen so that 1) this bad mv won't affect other non-dependent MVs; 2) user can get meaningful error message when using the bad MV.

Agree with the basic idea but this would become tricky with MV-on-MV. For example, MV2 is built on table T, an error in MV2 will also cause the failure of T, which look like an unexpected behavior. Although I don't have any better idea...

**Comment by hzxa21** [comment]
> Agree with the basic idea but this would become tricky with MV-on-MV. For example, MV2 is built on table T, an error in MV2 will also cause the failure of T, which look like an unexpected behavior. Although I don't have any better idea...

True. 

But in the current implementation with global checkpoint, downstream MV errors already unavoidably cause upstream MV/table failure, which means all dependent tables/MVs will be **freezed and become read-only**. Recovery may not neccessarily help in this case. 

This "frezze and read-only" behavior actually sounds reasonable to me because it can hardly be resolved without manual intervention. I am just thinking whether we should make this "freeze and read-only" behavior explicit to user plus we can skip checkpointing these freezed tables/MVs to minimize the impact of the failure.

**Pull request merged**